python/lsst/ip/isr/measureCrosstalk.py

```diff
@@ -316,17 +358,193 @@ def reduce(self, ratioList):
-        """Disable config output"""
-        return None
+    def measureCrosstalkCoefficients(self, ratios, rejIter=3, rejSigma=2.0):
+        """Measure crosstalk coefficients from the ratios.
 
-    def _getMetadataName(self):
-        """Disable metdata output"""
-        return None
+        Given a list of ratios for each target/source amp combination,
+        we measure a sigma clipped mean and error.
+
+        The coefficient errors returned are the standard deviation of
+        the final set of clipped input ratios.
+
+        Parameters
+        ----------
+        ratios : `list` of `list` of `numpy.ndarray`
+           Matrix of arrays of ratios.
+        rejIter : `int`
+           Number of rejection iterations.
+        rejSigma : `float`
+           Rejection threshold (sigma).
+
+        Returns
+        -------
+        coeff : `numpy.ndarray`
+            Crosstalk coefficients.
+        coeffErr : `numpy.ndarray`
+            Crosstalk coefficient errors.
+        coeffNum : `numpy.ndarray`
+            Number of pixels for each measurement.
+
+        Notes
+        -----
+        This has been moved into MeasureCrosstalkTask to allow for easier
+        debugging.
+
+        The lsstDebug.Info() method can be rewritten for __name__ =
+        `lsst.ip.isr.measureCrosstalk`, and supports the parameters:
+
+        debug.display['measure'] : `bool`
+            Display a histogram of the combined ratio measurements for
+            a pair of source/target amplifiers from the final set of
+            clipped input ratios.
+        """
+        if rejIter is None:
+            rejIter = self.config.rejIter
+        if rejSigma is None:
+            rejSigma = self.config.rejSigma
+
+        numAmps = len(ratios)
+        assert all(len(rr) == numAmps for rr in ratios)
+
+        coeff = np.zeros((numAmps, numAmps))
+        coeffErr = np.zeros((numAmps, numAmps))
+        coeffNum = np.zeros((numAmps, numAmps), dtype=int)
+
+        for ii, jj in itertools.product(range(numAmps), range(numAmps)):
+            if ii == jj:
+                values = [0.0]
+            else:
+                values = np.array(ratios[ii][jj])
+                values = values[np.abs(values) < 1.0]  # Discard unreasonable values
+
+            coeffNum[ii][jj] = len(values)
+
+            if len(values) == 0:
+                coeff[ii][jj] = np.nan
+                coeffErr[ii][jj] = np.nan
+            else:
+                if ii != jj:
+                    for rej in range(rejIter):
+                        lo, med, hi = np.percentile(values, [25.0, 50.0, 75.0])
+                        sigma = 0.741*(hi - lo)
+                        good = np.abs(values - med) < rejSigma*sigma
+                        if good.sum() == len(good):
+                            break
+                        values = values[good]
+
+            coeff[ii][jj] = np.mean(values)
+            coeffErr[ii][jj] = np.nan if coeffNum[ii][jj] == 1 else np.std(values)
```

Comment:
Confused by a couple of things here. You've tossed all the 1's already, so I don't think `coeffNum[ii][jj] == 1` can ever be True. Also, don't you mean zero anyway here?

Reply:
`coeffNum[ii][jj]` is the number of samples for this entry in the CT matrix.  Setting the error to `np.nan` shows that this isn't well estimated, instead of `np.std() = 0.0`, which would indicate it's perfectly estimated.